impress-org/givewp-next-gen
# Refactor: replace form builder build process with laravel mix

## Description



This PR replaces the form builder's build process from using react-scripts to laravel mix.

We now have 3 new npm scripts that refer to the form-builder scripts that we can call from the plugin root:

```
npm run dev:form-builder
npm run watch:form-builder
npm run build:form-builder
```

The below will be used just for the plugin assets.  The exception being `npm run build` which builds both the plugin and form-builder into production code.
```
npm run dev
npm run watch
npm run build
```

Also as a bonus we now have a new alias for the root of the form builder src file `@givewp/form-builder` so we can do things like:

```
import type {Block} from '@givewp/form-builder/types';
```

Along with the new build process, the form builder has been updated to dequeue as much baggage (scripts & styles) from WordPress as possible. 



## Affects



The form builder build process.

## Visuals



N/A

## Testing Instructions



Start in the root plugin and run `npm install`.  Then run `npm run build` and make sure everything works properly.
You can now test out `npm run dev:form-builder` and `npm run watch:form-builder`

## Pre-review Checklist
